### DISCUSSION TITLE
Arm64 docker images?

### DISCUSSION THREAD

**Discussion by badsyntax** [plausible/analytics#1286] [Anything Else]
I've been trying to run plausible with docker on my Apple M1 and I'm getting the following error from Docker Swarm: `unsupported platform on 1 node` On further inspection it looks like the [`plausible/analytics`](https://hub.docker.com/r/plausible/analytics) docker image does not support arm64.

Can arm64 version of the  docker images be published to docker hub?

**Comment by badsyntax** [comment]
I think the only change required to support this is to add `platforms: linux/amd64,linux/arm64` to the `build-push-action` here: https://github.com/plausible/analytics/blob/01412d090f7e0d319bc82dce565baa83161e167d/.github/workflows/docker.yml#L35

See: https://github.com/docker/build-push-action#customizing

**Comment by ukutaht** [comment]
So the reason we don't provide ARM images is because Clickhouse doesn't support ARM (https://clickhouse.tech/docs/en/operations/requirements/) so we thought there's no point. Plausible is pretty useless without the database.

Are you planning to run the database remotely? I don't think you can run Clickhouse on M1

**Comment by badsyntax** [comment]
Thanks @ukutaht, I wasn't aware clickhouse don't support ARM.  This is rather annoying. I might create an issue in their repo.

**Reply by onthebridgetonowhere** [reply]
You could build the ARM images yourself, it is not too complicated. If you'd like to try, I wrote a [blog post on how to self host](https://stefanstanciulescu.com/blog/plausible-analytics-on-raspberry-pi/) it on a Raspberry Pi 3.

**Reply by ukutaht** [reply]
Nice, thanks @onthebridgetonowhere !

**Comment by FarisZR** [comment]
Clickhouse now supports ARM

I created a PR, with test builds for the self-hosted and master versions.
https://github.com/plausible/analytics/pull/2103

**Reply by DennisLoska** [reply]
Thanks, it's working for me on Raspberry Pi 4. Though I disabled the mail service since I don't need it.

**Comment by ruslandoga** [comment]
👋

[`v1.5.0-rc.0`](https://github.com/plausible/analytics/discussions/2324) is out with arm64 support! Please try it out :)
